variables.tf

```diff
@@ -123,15 +124,37 @@ This object describes the subnets to create within the virtual network.
 
   Example Inputs:
 
-  ```hcl
-  subnets = {
-    "CoreSubnet" = {
-      address_prefixes                = ["100.0.1.0/24"]
-      default_outbound_access_enabled = false
-      delegate_to                     = "Microsoft.ContainerInstance/containerGroups"
+```hcl
+subnets = {
+  "CoreSubnet" = {
+    address_prefixes                = ["100.0.1.0/24"]
+    default_outbound_access_enabled = false
+  }
+  "DevopsSubnet" = {
+    address_prefixes                = ["100.0.2.0/24"]
+    default_outbound_access_enabled = false
+    delegate_to                     = "Microsoft.ContainerInstance/containerGroups"
+    create_network_security_group   = true
+  }
+  "ToolingSubnet" = {
+    address_prefixes                = ["100.0.3.0/24"]
+    default_outbound_access_enabled = false
+    create_network_security_group   = true
+    network_security_group_config = {
+      azure_default = true
     }
   }
-  ```hcl
+  "OtherSubnet" = {
+    address_prefixes                = ["100.0.4.0/24"]
+    default_outbound_access_enabled = false
+    no_nsg_association              = true
+  }
+  "AzureBastionSubnet" = {

```

Comment:
I'm wondering if some of the stuff in this module related to the Bastion should not be made part of the bastion mcaf module instead.

Any particular reason why we don't create the necessary subnet there? I guess that might be related to your address_prefixes?

Reply:
imo, you should create the networks from your networking module.
if you decide to not use the networking module, that up-to-you ofc.
but then you indeed need to split all your network creations from other locations.
but you should be able to deploy this from the networking module, thats its purpose.

Reply:
haven't checked that module, but yes we could make a not that you could use this module to create it.
but ofc it should not be required, people are still allowed to generate subnets elsewehere right?

Reply:
yeah just so people know that they need to do more than just use the bastion module in order for them to deploy a bastion